### PR TITLE
feat(frontend): Stateful v2 Node styling. Fix #6280

### LINKED ISSUE #6280: Define static status css styling for Execution, Artifact.

### PR DESCRIPTION
**Pull request by zijianjoy**
**Description of your changes:**

Fix #6280
Partial #6419

statefulnodes3

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by Bobgy** [comment]
Discussion: Shall we use the same icons as KFP v1 for the states that exist in v1? I think that's most friendly to existing users.

File: frontend/src/components/graph/ArtifactNode.tsx [hunk 1]
```diff
@@ -62,3 +62,27 @@ function ArtifactNode({ id, data }: ArtifactNodeProps) {
 }
 
 export default ArtifactNode;
+
+function getIcon(state: Artifact.State | undefined) {
+  if (state === undefined) {
+    return (
+      <div className='px-2 flex flex-col justify-center items-center rounded-l-lg'>
```
Comment (Bobgy) [inline frontend/src/components/graph/ArtifactNode.tsx:69]: All the class names are the same, maybe give them a name and reuse?

File: frontend/src/components/graph/Constants.ts [hunk 2]
```diff
@@ -22,3 +24,11 @@ export interface SubDagFlowElementData extends FlowElementDataBase {
   // Callback action if a SubDag expand button is clicked.
   expand: (nodeKey: string) => void;
 }
+
+export interface ExecutionFlowElementData extends FlowElementDataBase {
```
Comment (Bobgy) [inline frontend/src/components/graph/Constants.ts:28]: nit: This is purely a data type.
Usually I use typescript type syntax for data types and only use interface for sth that can be a class.
You can do the following:
```
export type A = B & C
```
It means A type has all the fields from B type and C type.

File: frontend/src/components/graph/ExecutionNode.tsx [hunk 3]
```diff
@@ -30,19 +37,26 @@ import { Handle, Position } from 'react-flow-renderer';
 //   SKIPPED = 'SKIPPED',
 //   NOT_READY = 'NOT_READY', // TBD: Represent QUEUED, NOT_TRIGGERED and UNSCHEDULABLE
 // }
-
-export interface ExecutionNodeData {
-  label: string;
-}
+// enum State {
```
Comment (Bobgy) [inline frontend/src/components/graph/ExecutionNode.tsx:40]: Clean up?

**Review by Bobgy — COMMENTED** [review]
/lgtm
The implementation LGTM.
I would defer to ask designer for the exact choice of icon and color .

**Comment by zijianjoy** [comment]
> Discussion: Shall we use the same icons as KFP v1 for the states that exist in v1? I think that's most friendly to existing users.

That is a good point, however, I would say that the state enum information is a bit different between argo and IR. Therefore, I am leaning towards the consistency between KFPv2 and MP, other than KFPv1 and KFPv2. We can help resolve the confusion by showing the node state on side panel.

File: frontend/src/components/graph/ArtifactNode.tsx (hunk 1 above)
Comment (zijianjoy) [reply frontend/src/components/graph/ArtifactNode.tsx:69]: Awesome point! Added a function for the wrapper div.

File: frontend/src/components/graph/Constants.ts (hunk 2 above)
Comment (zijianjoy) [reply frontend/src/components/graph/Constants.ts:28]: I have been thinking about interface vs type for some time myself as well. Based on https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces, I think both are very similar, and `type` is more strict from the `extension` capability. Sounds good on using `type` here.

File: frontend/src/components/graph/ExecutionNode.tsx (hunk 3 above)
Comment (zijianjoy) [reply frontend/src/components/graph/ExecutionNode.tsx:40]: Done.

**Comment by zijianjoy** [comment]
> /lgtm
> The implementation LGTM.
> I would defer to ask designer for the exact choice of icon and color .

Thank you so much! I have consulted with Weikun and updated the screenshot as the result of discussion.

**Pull request merged**